### PR TITLE
added postTag provider

### PR DESCRIPTION
**Pull request by FORDBA**
# Description
Please include a summary of the change and which issue is fixed / implemented. Please also include relevant motivation and context. List any dependencies that are required for this change (e.g., do you need to run an npm install to test?).
Fixes # (issue) # (issue2) ... (these should be the numbers of issues we have created / on the project board for this project)
## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
# How Has This Been Tested?
Please describe step-by-step the process that a reviewer can follow to adequately test this PR. 
- [ ] Check PostTagProvider code for errors

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by mgjeffries — APPROVED** [review]
Great!  

... Whoops! Missed some things

File: src/components/posttags/PostTagProvider.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import React, { useState, createContext } from "react";
+
+export const PostContext = createContext();
```
Comment (mgjeffries) [inline src/components/posttags/PostTagProvider.js:3]: This should be called PostTagContext

**Review by mgjeffries — CHANGES_REQUESTED** [review]
Context needs to be re-named

File: src/components/posttags/PostTagProvider.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import React, { useState, createContext } from "react";
+
+export const PostTagContext = createContext();
+
+export const PostTagProvider = props => {
+    const [thisPostTags, setPostTags] = useState([]);
```
Comment (mgjeffries) [inline src/components/posttags/PostTagProvider.js:6]: Remove notion of thisPostTags

File: src/components/posttags/PostTagProvider.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+
+    const getPostTagsByPostId = postId => {
+        return fetch(`http://localhost:8088/post_tags?post_id=${postId}`)
+            .then(res => res.json())
+            .then(setPostTags);
```
Comment (mgjeffries) [inline src/components/posttags/PostTagProvider.js:12]: Remove notion of thisPostTags

File: src/components/posttags/PostTagProvider.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    const addPostTag = postTag => {
+        return fetch("http://localhost:8088/post_tags", {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json"
+            },
+            body: JSON.stringify(postTag)
+        })
+            .then(getPostTagsByPostId)
```
Comment (mgjeffries) [inline src/components/posttags/PostTagProvider.js:23]: No need to call getPostTagsByPostId here

File: src/components/posttags/PostTagProvider.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    const deletePostTag = (postTagId) => {
+        return fetch(`http://localhost:8088/post_tags/${postTagId}`, {
+            method: "DELETE"
+        })
+            .then(getPostTagsByPostId)
```
Comment (mgjeffries) [inline src/components/posttags/PostTagProvider.js:30]: No need to call getPostTagsByPostId here

**Review by mgjeffries — CHANGES_REQUESTED** [review]

**Review by gavinschriver — APPROVED** [review]
this is definitely a thing that looks like stuff

**Review by gavinschriver — APPROVED** [review]

**Review by mgjeffries — APPROVED** [review]
Thanks!

**Pull request merged**